sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -2011,18 +2040,34 @@ def task_worker(self, stopping_event, sfp_error_event):
 
         helper_logger.log_info("Stop SFP monitoring loop")
 
-    def task_run(self, sfp_error_event):
+    def run(self):
+        self.thread_id = threading.current_thread().ident
         if self.task_stopping_event.is_set():
             return
-
-
-        self.task_process = multiprocessing.Process(target=self.task_worker, args=(
-            self.task_stopping_event, sfp_error_event))
-        self.task_process.start()
-
-    def task_stop(self):
+        try:
+            self.task_worker(self.task_stopping_event, self.sfp_error_event)
+        except Exception as e:
+            helper_logger.log_error("Exception occured at {} thread due to {}".format(threading.current_thread().getName(), repr(e)))
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+            msg = traceback.format_exception(exc_type, exc_value, exc_traceback)
+            for tb_line in msg:
+                for tb_line_split in tb_line.splitlines():
+                    helper_logger.log_error(tb_line_split)
+            self.exc = e
+            self.main_thread_stop_event.set()
+
+    def raise_exception(self):
```

Comment:
can you add comment here why this exception is raised here?

Reply:
I have added a comment now.